lamindb/_feature.py

```diff
@@ -16,25 +19,46 @@
 if TYPE_CHECKING:
     from lnschema_core.types import FieldAttr
 
-FEATURE_TYPES = {
-    "number": "number",
-    "int": "int",
-    "float": "float",
-    "bool": "bool",
-    "date": "date",
-    "datetime": "datetime",
-    "str": "cat",
-    "object": "cat",
+
+FEATURE_DTYPES = {
+    "cat",  # categorical variables
+    "num",  # numerical variables
+    "str",  # string variables
+    "int",  # integer variables
+    "float",  # float variables
+    "bool",  # boolean variables
+    "date",  # date variables
+    "datetime",  # datetime variables
+    "object",  # this is a pandas type, we're only using it for complicated types, not for strings
 }
 
 
-def convert_numpy_dtype_to_lamin_feature_type(dtype, str_as_cat: bool = True) -> str:
-    orig_type = dtype.name
-    # strip precision qualifiers
-    type = "".join(i for i in orig_type if not i.isdigit())
-    if type == "object" or type == "str":
-        type = "cat" if str_as_cat else "str"
-    return type
+def get_dtype_str_from_dtype(dtype: Any) -> str:
+    if not isinstance(dtype, list) and dtype.__name__ in FEATURE_DTYPES:
+        dtype_str = dtype.__name__
+    else:
+        error_message = "dtype has to be of type Record or list[Record]"
+        if isinstance(dtype, Record):
+            dtype = [dtype]
+        elif not isinstance(dtype, list):
+            raise ValueError(error_message)
+        registries_str = ""
+        for registry in dtype:
+            if not hasattr(registry, "__get_name_with_schema__"):
+                raise ValueError(error_message)
+            registries_str += registry.__get_name_with_schema__() + "|"
+        dtype_str = f'cat[{registries_str.rstrip("|")}]'
+    return dtype_str
+
+
+def convert_pandas_dtype_to_lamin_dtype(pandas_dtype) -> str:
+    if is_string_dtype(pandas_dtype) and not isinstance(pandas_dtype, CategoricalDtype):
+        dtype = "str"
+    else:
+        # strip precision qualifiers
+        dtype = "".join(i for i in pandas_dtype.name if not i.isdigit())
```

Comment:
```suggestion
        dtype = "".join(char for char in pandas_dtype.name if not char.isdigit())
```
Nit

Reply:
Absolutely right... that's horrible; adopting your suggestion locally.